### DISCUSSION TITLE
The filter types for requestDevice (and requestLEScan) don't seem to match the spec

### DISCUSSION THREAD

**Discussion by ghost** [capacitor-community/bluetooth-le#430] [Ideas]
Looking at https://webbluetoothcg.github.io/web-bluetooth/#device-discovery I noticed that `requestDevice` (and `requestLeScan`) don't seem to match the same filter types as the spec. It's also the case that `@types/webbluetooth` is included as a dependency, but specifically `BluetoothDataFilterInit` isn't used as the base for filters.

Should we align more with that?

**Comment by pwespi** [comment]
Good question. Initially I was thinking about trying to implement the WebBluetooth spec with this plugin. But I decided that the cost was too high and that there are things that are possible in iOS and specifically in Android that are not in WebBluetooth, so anyway the plugin API will be different from the WebBluetooth API. Therefore I decided not to follow the WebBluetooth spec too closely.

The @types/webbluetooth is used in the web part of the plugin, that's why it is included.

Regarding scanning filters, one thing you can do is use `requestLEScan` and then filter more in your application code before showing the results to the user.

Do you have a use case where having `BluetoothDataFilterInit` in the API would be good? Personally I never had the need for it.

**Comment by ghost** [comment]
I think we should adhere to it as much as it makes sense to. The design for the filters for example is pretty decent, and maps to the options android already has (for the most part.

I personally need to use the manufacturer data filter. I've got it somewhat working already if i can figure out how to read the options in kotlin.  I don't quite know how to deal with js objects on the kotlin side. 

why not move the android (and/or ios if there are any) options into a separate key like `android: { key: "value", key2: "value2"}`? and the the top level values can be mostly webbluetooth?

One advantage of that is that folks who wanna write some generic bluetooth code can even use the webbluetooth interface provided by libraries like noble in nodejs as well without having too much of a separate wrapping for that.  This way one could use this plugin's interface when using the electron capacitor integration. I don't think it'd take much work to enable that.

Then we could have `noble: { key: "value"}` for that as well.
